### PR TITLE
[WIP][BUGFIX] Resolving typo3cms path if composer not used (#446)

### PR DESCRIPTION
**Pull request by tomasnorre**

### CODE REVIEW

**Comment by tstahn** [comment]
Just a thought/general question after I've checked on these changes:

Is there any reason why we are not dropping the typo3-console usage/dependency completely in favor of the TYPO3 Cores `typo3/sysext/core/bin/typo3` binary?

As I see it this would have several benefits:
- no composer setup required
- no typo3-console required
- native TYPO3 support (no 3rd party dependencies)
- low threshold for (new )users as there is no need to learn about typo3-console first
- no need for path mapping/finding inside the crawler extension
- no need for checking whether or not crawler is used in composer environment

Last but not least:
- less overhead
- less potential for things to break

**Comment by tomasnorre** [comment]
@tstahn 

I agree on the points here. 
But would not like to switch in the 6.x versions. 

I would prefer only doing this switch in the V9. 
Don't want to introduce to0 many new things in a version, that will soon be obsolete.

But I'll consider it, because you're totally right on your point, and if someone decides to use the typo3cms (typo3-console) it still working, but not a hard dependency anymore..

But we cannot do that because of symfony/console before we drop support for TYPO3V7, that's also one of my reason to only do it for the TYPO3V9 version.

File: Classes/Service/ProcessService.php [hunk 1]
```diff
@@ -225,20 +226,26 @@ public function startProcess()
     public function getCrawlerCliPath()
     {
         $composerRootDir = getenv('TYPO3_PATH_COMPOSER_ROOT') . '/';
-        $jsonDecoded = json_decode(file_get_contents($composerRootDir . 'composer.json'), true);
+        $composerFile = $composerRootDir . 'composer.json';
+        $phpPath = $this->crawlerController->extensionSettings['phpPath'] . ' ';
+        $cliPart = 'typo3cms crawler:crawlqueue';
+
+        if (file_exists($composerFile)) {
+            $jsonDecoded = json_decode(file_get_contents($composerRootDir . 'composer.json'), true);
```
Comment (tstahn) [inline Classes/Service/ProcessService.php]: You could have re-used `$composerFile` here.

Comment (tomasnorre) [reply Classes/Service/ProcessService.php]: Sure

**Pull request merged**